### PR TITLE
typeKeywords for older groups

### PR DESCRIPTION
**Pull request by jmhauck**
#765

@MikeTschudi one spot I missed. Older group templates won't have the typeKeywords array.

### CODE REVIEW

File: packages/deployer/src/deploySolutionFromTemplate.ts [hunk 1]
```diff
@@ -347,6 +335,33 @@ export function deploySolutionFromTemplate(
...
+  templates: common.IItemTemplate[]
+): common.IItemTemplate[] {
+  return templates.map(
+    (template: any) => {
+      const sourceId: string = "source-" + template.itemId;
+      /* istanbul ignore else */
+      if (template.item) {
+        if (template.item!.typeKeywords) {
+          template.item!.typeKeywords!.push(sourceId);
+        } else /* istanbul ignore else */ if (common.getProp(template, "item.type") === "Group") {
```
Comment (MikeTschudi) [inline packages/deployer/src/deploySolutionFromTemplate.ts:354]: Is it necessary to test for the group type? This clause could simply handle any item/group that doesn't have `typeKeywords` for any reason.

**Review by MikeTschudi — APPROVED** [review]

File: packages/deployer/src/deploySolutionFromTemplate.ts (hunk 1 above)
Comment (jmhauck) [reply packages/deployer/src/deploySolutionFromTemplate.ts:354]: @MikeTschudi I wasn't sure if some item types could/would be expected to not have typeKeywords.

Comment (jmhauck) [reply packages/deployer/src/deploySolutionFromTemplate.ts:354]: Happy to update if you think it's best though

Comment (MikeTschudi) [reply packages/deployer/src/deploySolutionFromTemplate.ts:354]: The REST API documentation doesn't say what happens when there are no typeKeywords, so this struck me as a convenient way to survive the API not returning an empty list when there are no typeKeywords.

**Pull request merged**